Update ont-zte-f601.md (#181)
hack-gpon/hack-gpon.github.io: _ont/ont-zte-f601.md

Where `X` can be `0/1` based on the image you want to boot.

> You can also clone currently running image into other slot using this command:

Reviewer: ```suggestion
You can also clone the currently running image into other slot using this command:
```

```sh
syn_version
```

# Use

> {% include alert.html content="Commands have been tested on V6/V7 HW rev on TIM and OF firmware" alert="Note"  icon="svg-info" color="blue" %}

Reviewer: ```suggestion
{% include alert.html content="Commands have been tested on V6/V7 HW rev. on TIM and OpenFiber firmwares" alert="Note"  icon="svg-info" color="blue" %}
```

## Enable Telnet

> {% include alert.html content="This is an external script ([ZTE Telnet enabled](https://github.com/douniwan5788/zte_modem_tools)), so use at your own risk! Credential doesn't survive at reboot!" alert="Note"  icon="svg-info" color="blue" %}

Reviewer: ```suggestion
{% include alert.html content="This is an external script ([ZTE Telnet enabled](https://github.com/douniwan5788/zte_modem_tools)), so use it at your own risk! Credential doesn't survive at reboot!" alert="Note"  icon="svg-info" color="blue" %}
```

> {% include alert.html content="For Italian users, it only works on versions V6.0.10N40 (TIM) and V6.0.10P6N7 (OpenFiber)" alert="Note"  icon="svg-info" color="blue" %}

Reviewer: ```suggestion
{% include alert.html content="For italian users, it only works on versions V6.0.10N40 (TIM) and V6.0.10P6N7 (OpenFiber)" alert="Note"  icon="svg-info" color="blue" %}
```

```sh
python3 zte_factroymode.py --user admin --pass admin --ip 192.168.1.1 --port 80 telnet open
```

You should get this output and credentials to login over telnet:

```sh
trying  user:"admin" pass:"admin"
reset facTelnetSteps:
reset OK!

facStep 1:
OK!

facStep 2:
OK!

facStep 3:
OK!

facStep 4:
OK!

facStep 5:
OK!

done
Username: 2W3iqFVt
Password: Eqb8X8Qt
```

## Enable console redirection

> To see omcidebug messages on Telnet you need to perform this command (just first time of each connection):

Reviewer: ```suggestion
To see omcidebug messages on Telnet you need to execute this command (just the first time of each connection):
```

---------------------------------------------------------------------
```

## Querying a particular OMCI ME

```sh
sendcmd 132 omcidebug showmedata ID_MIB (eg. 7 for Firmware version)
```

> This command will print out the result like this one:

Reviewer: ```suggestion
This command will print out a result like this one:
```

```sh

##################################
MIB INFO:
         ME CLASS: 7
         DB NAME: soft_image, DBHandle: 14
##################################

<-----MeID[ 0x0000,0 ], Addr[ 0x19a011]----->
          Version:V6.0.10N41
     Is committed:01
        Is active:01
         Is valid:01

<-----MeID[ 0x0001,1 ], Addr[ 0x19a031]----->
          Version:V6.0.10N39
     Is committed:00
        Is active:00
         Is valid:01
---------------------------------------------------------------------
```

# GPON/OMCI settings

## Setting ONU GPON Serial Number

> {% include alert.html content="You have to change S/N and the VID. 2176 is for the VID (first 4 letters of the S/N) and 2177 is for the last 8 digits" alert="Note"  icon="svg-info" color="blue" %}

Reviewer: ```suggestion
{% include alert.html content="You have to change S/N and the VID. 2176 is for the VID (first 4 letters of the S/N) and 2177 is for the last 8 digits of the S/N" alert="Note"  icon="svg-info" color="blue" %}
```

{% include alert.html content="The PLOAM password is stored in the ASCII format." alert="Note"  icon="svg-info" color="blue" %}
This can be done easily via web ui. If you prefer to do it via the shell use:
```sh
setmac 1 2181 1234567890
setmac 1 2178 1234567890
```

## Change ONU HW\SW Version and Permanent TELNET 

{% include alert.html content="The only way to change HW\SWVer on this ONT is to modify the firmware, so do it at your own risk" alert="Note"  icon="svg-info" color="blue" %}
{% include alert.html content="This procedure was only tested on TIM V6.0.10N40 and OF V6.0.10P6N7 firmwares" alert="Note"  icon="svg-info" color="blue" %}
{% include alert.html content="This procedure work with `ZTE_Firmware_Mod.py` v1.0.0" alert="Note"  icon="svg-info" color="blue" %}

Needed tools:

- Linux VM or WSL with Python >3.3
- [ZTE Telnet enabled](https://github.com/douniwan5788/zte_modem_tools)
- [ZTE Firmware Mod Script](http://github.com/hack-gpon/ZTE-firmware-mod)
- TFTP server

Download the script `ZTE_Firmware_Mod.py` and place in the same folder where you have the `kernel0` or `kernel1` mtd dump taken from step `**Backup ONT Paritions for HW\SW Version Mod**`.

> Run the script with the following parameters, you can use `-h` for help. In this example we are just replace firmware version with `V6.0.10N40`. You can put your own version here, maximium 15 characters, this parameter is mandatory:

Reviewer: ```suggestion
Run the script with the following parameters, you can use `-h` for help. In this example we are just replacing the firmware version with `V6.0.10N40`. You can put your own version here, maximium 15 characters. This parameter is mandatory:
```

> If you have create partition dump with different name, please put the correct name instead of `kernel0`

Reviewer: ```suggestion
If you need to create a partition dump with a different name, please put the correct name instead of `kernel0`
```

Please refer to Hack-GPON Wiki for how-to: https://hack-gpon.github.io/ont-zte-f601/
---------------------------------------
To proceed please enter 'y', otherwise 'n' to exit: y

---------------------------------------
Step 1: Patching zImage and fix uImage Header..
------: Done in 4.846 secs
Step 2: Add back ZTE Header and Firmware Version..
------: Old FW version V6.0.10N39
------: New FW version V6.0.10N40
------: Done in 0.008 secs
Step 3: Write firmware file..
------: Done in 0.003 secs

---------------------------------------
How to flash:

Copy firmware file fw_mod.bin into your TFTP server and flash is using this procedure on the ONT over telnet:

cd /var/tmp
tftp -l fw.bin -r fw_mod.bin 192.168.1.100 -g
fw_flashing -d 0 -r 0 -c 1 -f fw.bin

After you get prompt back, erase old configurations:

rm /userconfig/cfg/*.xml

Create dummy files for HW\SWVer spoofing:
!!! CHANGE IT BASED ON YOUR ORIGINAL ONT !!!
echo V6.0 > /userconfig/cfg/hwver
echo V6.0.10N40 > /userconfig/cfg/swver

Then run these commands to switch software bank and reboot the ONT:

upgradetest switchver
reboot
---------------------------------------
Good luck!
```

**Two last steps!**

> If you are swapping from TIM to OpenFiber Firmware, or viceversa, before reboot the ONT you have to run these two command based on the firmware version:

Reviewer: ```suggestion
If you are swapping from TIM to OpenFiber Firmware, or viceversa, you have to run these two command before rebooting the ONT based on the firmware version:
```

```sh
tftp -l kernel1 -r kernel1 -p 192.168.1.X (where X is the IP of your PC)
```

Delete dump

```sh
 rm kernel1
```

**Dump mtd4 (others)**

```sh
 cat /dev/mtd4 > others
```

Copy the dumped firmware via TFTP to you VM or Windows machine with this commnad:

```sh
tftp -l others -r others -p 192.168.1.X (where X is the IP of your PC)
```

Delete dump

```sh
 rm others
```

**Dump mtd5 (param_tags)**

```sh
 cat /dev/mtd5 > param_tags
```

Copy the dumped firmware via TFTP to you VM or Windows machine with this commnad:

```sh
tftp -l param_tags -r param_tags -p 192.168.1.X (where X is the IP of your PC)
```

Delete dump

```sh
 rm param_tags
```

**Dump mtd6 (usercfg)**

```sh
 cat /dev/mtd6 > usercfg
```

Copy the dumped firmware via TFTP to you VM or Windows machine with this commnad:

```sh
tftp -l usercfg -r usercfg -p 192.168.1.X (where X is the IP of your PC)
```

Delete dump

```sh
 rm usercfg
```

## Change region code

{% include alert.html content="Looks like TIM and OF firmwares work only with their stock factory conf, so 97 or 116, otherwise no PPPoE" alert="Note"  icon="svg-info" color="blue" %}

> ZTE has create various region code that loads default valuse based on local ISP, this configuration can be changed using this command:

Reviewer: ```suggestion
ZTE has created various region codes that load default valuse based on the local ISP. This configuration can be changed using this command:
```
